sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,24 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+def reduce_boolean_expr(boolean_expr,symbols=[]):
+    boolean_expr=to_cnf(boolean_expr)
+    if boolean_expr.has(And) and not(boolean_expr.has(Or)):
+          return(reduce_inequalities(boolean_expr.args,symbols))
+    elif boolean_expr.has(Or):
```

Comment:
So now we have an And with one or more Or args, a flat Or, or an Or with one ore more And args.

Reply:
I don't think `has(Or)` or `has(And)` is the right thing to check here because that can pick up subexpressions that contain `Or/And`. Really this should have a case for handling if the expression *is* an `Or` and another for `And` etc.

Reply:
Well, line 993 has been changed to isinstance as suggested by @smichr . Is there something additional to that that need to be changed?

Reply:
It should be testing more precisely that it is an `And` of relationals rather than using `not expr.has(Or)`.

Reply:
I am leaving a suggestion on the issue page.